dev/src/main/java/dataInput/WikiArticleLoader.java

```diff
@@ -0,0 +1,26 @@
+package dataInput;
+
+import java.io.IOException;
+import java.util.MissingResourceException;
+import java.util.Set;
+
+public interface WikiArticleLoader {
+    /**
+     * Downloads the article from Wikipedia with title @title in the language @language and returns it as a @{@link WikiArticle}
+     * @param title the title of the article
+     * @param language the language of the article
+     * @return a @{@link WikiArticle}
+     * @throws MissingResourceException if the article was not available
+     */
+    WikiArticle loadArticle(String title, WikiArticle.Language language) throws MissingResourceException, IOException;
+
+
+    /**
+     * Downloads the articles from Wikipedia with title @title in the languages @languages and returns it as a set of @{@link WikiArticle}
+     * @param title the title of the article
+     * @param languages the languages of the article
+     * @return a set containing the @{@link WikiArticle}s. If an article cannot be found in the specific language it'll be ignored
+     */
+    Set<WikiArticle> loadArticleInMultipleLanguages(String title, Set<WikiArticle.Language> languages);
```

Comment:
I expect this to work better or worse depending on how you will implement it: If you search for the same title in different languages, you will undoubtedly get a lot of misses save the cases in which titles are named entities; if you do the equivalent of following the language links on a Wikipedia HTML page, you might have a chance to get a lot of articles in different languages. I find it difficult to estimate however how feasible that second option really is.

Reply:
true point! now deleted